### PR TITLE
Fix/settings button

### PR DESCRIPTION
**Pull request by olell**
- Fixed bug (maybe only on iOS) that the FAB was not clickable
- Changed settings modal from react-native to react-native-paper
- Added checkbox for secure connection (ws/wss http/https) in settings modal
- AsyncStorage for secureConnection flag
- Changed to stored serverHost and secureConnection instead of hardcoded api/socket addresses

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -65,12 +82,17 @@ function App(): JSX.Element {
...
     }
-    if (socketRef.current == null) {
-      socketRef.current = io('wss://olel.de', { transports: ['websocket'] });
-    }
     return () => {};
   }, []);
 
+  useEffect(() => {
+    if (socketRef.current == null || !socketRef.current.connected) {
+      socketRef.current = io(getWSUrl(serverHost, secureConnection), {
```
Comment (mowoe) [inline App.tsx:90]: serverHost is initially undefined, i think this needs to be handled here

File: App.tsx [hunk 2]
```diff
@@ -48,13 +55,23 @@ function App(): JSX.Element {
...
         setServerHost(e);
+      } else {
+        setServerHost('olel.de'); // Fallback default ?
+      }
+    });
+    getSecureConnection(e => {
+      if (e) {
+        setSecureConnection(e);
+      } else {
+        setSecureConnection(false);
```
[the remark this answers is not in the document]
Comment (olell) [reply App.tsx:74]: e is boolean or null and the setter only accepts boolean values. Is there any better solution to set it to false if e is none?

File: components/globalSettings.tsx [hunk 3]
```diff
@@ -1,50 +1,72 @@
...
+          onPress={() => {
+            setSecureConnection(!secureConnection);
+          }}
+        />
+        <Button
+          mode="contained"
+          style={{ marginTop: 30 }}
+          onPress={() => {
+            setUrl(hostInput);
+            storeSecureConnection(secureConnection);
```
[the remark this answers is not in the document]
Comment (olell) [reply components/globalSettings.tsx:63]: Yes good point! I'll take care of it

File: App.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mowoe) [reply App.tsx:74]:
```js
const foo = undefined;
booleanSetter(!!foo)
```

[the remark this answers is not in the document]
Comment (olell) [reply App.tsx:74]: Oh okay, that's much better :D

File: App.tsx [hunk 4]
```diff
@@ -48,13 +55,23 @@ function App(): JSX.Element {
...
   const scrollViewRef = useRef<ScrollView | null>(null);
   const theme = useTheme();
 
   useEffect(() => {
     getUrl(e => {
       if (e) {
         setServerHost(e);
+      } else {
+        setServerHost('olel.de'); // Fallback default ?
+      }
```
Comment (mowoe) [inline App.tsx:68]:
```suggestion
    setServerHost(e ? e : 'olel.de');
```

File: components/globalSettings.tsx [hunk 5]
```diff
@@ -64,6 +86,24 @@ export const setUrl = (url: string) => {
     });
 };
 
+export const getSecureConnection = (callback: (e: boolean | null) => void) => {
```
Comment (mowoe) [inline components/globalSettings.tsx:89]: I would move this. Multiple exports in component modules is a bit confusing imho. These are utility functions.

**Review by mowoe — APPROVED** [review]
lgtm!

**Pull request merged**